api/pacman-api-admin/src/main/java/com/tmobile/pacman/api/admin/repository/service/ContrastAccountServiceImpl.java

```diff
@@ -0,0 +1,238 @@
+    }
+
+    @Override
+    public String serviceType() {
+        return CONTRAST;
+    }
+
+    @Override
+    public AccountValidationResponse validate(CreateAccountRequest accountData) {
+        AccountValidationResponse validationResponse = validateRequestData(accountData);
+        if (validationResponse.getValidationStatus().equalsIgnoreCase(FAILURE)) {
+            LOGGER.info("Validation failed due to missing parameters");
+            return validationResponse;
+        }
+        return getOrganizationName(accountData);
+    }
+
+    private AccountValidationResponse getOrganizationName(CreateAccountRequest pluginData) {
+        AccountValidationResponse validationResponse = new AccountValidationResponse();
+        validationResponse.setValidationStatus(FAILURE);
+        validationResponse.setMessage(VALIDATION_FAILED);
+        try {
+            String apiUrl = new URIBuilder(String.format(CONTRAST_URL_TEMPLATE, pluginData.getContrastEnvironmentName()))
+                    .setPath(ORGANIZATIONS_URL_PATH + pluginData.getContrastOrganizationId()).build().toString();
+            String authKey = Base64.getEncoder().encodeToString((pluginData.getContrastUserId() + ":"
+                    + pluginData.getContrastServiceKey()).getBytes());
+            HttpGet request = new HttpGet(apiUrl);
+            request.setHeader(HttpHeaders.AUTHORIZATION, authKey);
+            request.setHeader(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE);
+            request.setHeader("API-Key", pluginData.getContrastApiKey());
+            try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+                try (CloseableHttpResponse response = httpClient.execute(request)) {
+                    int statusCode = response.getStatusLine().getStatusCode();
+                    if (statusCode != 200) {
+                        String errorMessage = handleErrorResponse(response.getStatusLine().getStatusCode());
+                        validationResponse.setErrorDetails(errorMessage);
+                        return validationResponse;
+                    }
+                    JsonNode jsonResponse = objectMapper.readTree(response.getEntity().getContent());
+                    String organizationName = jsonResponse.get("name").textValue();
+                    if (StringUtils.isEmpty(organizationName)) {
+                        validationResponse.setErrorDetails("Couldn't read Organization name");
+                        return validationResponse;
+                    }
+                    pluginData.setContrastOrganizationName(organizationName);
+                    validationResponse.setValidationStatus(SUCCESS);
+                    validationResponse.setErrorDetails(null);
+                    validationResponse.setMessage("Connection established successfully");
+                    return validationResponse;
+                }
+            }
+        } catch (UnknownHostException e) {
```

Comment:
Single handler `(Exception e)` is pretty much enough. Just return message that validation is failed and log the rest.

Reply:
I was trying to give more specific message

Reply:
No need to do that - if execution reached to this point it means something completely messed up and validation didn't go as expected. Single exception that validation failed is enough. We can get all specific info from logs if we would like to.

Reply:
But there is chance that User will give wrong environment name and when it happens it will generate UnknownHostException